utils/fetch_release_data.py

```diff
@@ -0,0 +1,178 @@
+    gem_name = os.path.basename(repo_name)
+    gem_data_list = []
+    for rel in releases:
+        if rel.prerelease:  # ignore pre releases
+            continue
+        version = rel.tag_name.lstrip('v')
+        li = version.split('-')
+        if len(li) > 1:  # ignore alpha or beta releases
+            continue
+        gem_data = {}
+        rel_id = f"{gem_name}-{version}"
+        gem_data['id'] = rel_id
+        if rel_id in EXT_PARENT_ID:
+            gem_data['externalParentId'] = EXT_PARENT_ID[rel_id]
+        else:
+            gem_data['externalParentId'] = []
+        gem_data['releaseDate'] = rel.published_at.strftime("%Y-%m-%d")
+        gem_data['releaseLink'] = rel.html_url
+        if rel_id in RELEASE_PMID_DICT:
+            gem_data['PMID'] = RELEASE_PMID_DICT[rel_id]
+        elif gem_name in GEM_PMID_DICT:
+            gem_data['PMID'] = GEM_PMID_DICT[gem_name]
+        else:
+            gem_data['PMID'] = ''
+        gem_data['count'] = get_count(repo_name, git_api)
+        gem_data['validation'] = get_validation(repo_name, git_api)
+        gem_data_list.append(gem_data)
+
+    # sort the releases in the ascending order
+    gem_data_list = gem_data_list[::-1]
+
+    return gem_data_list
+
+
+def get_integrated_gems_list(repo_name, git_api):
+    """Get the list of integrated GEMS from Github"""
+    repo = git_api.get_repo(repo_name)
+    file_content = repo.get_contents("integrated-models/integratedModels.json")
+    data = json.loads(file_content.decoded_content.decode())
+    gems_list = []
+    for item in data:
+        gems_list.append(item['short_name'])
+    return gems_list
+
+
+def generate_data(git_api, basedir, is_dryrun):
+    """Fetch data from Github and output the result in JSON format"""
+    owner = "SysBioChalmers"
+    gems_list = get_integrated_gems_list("MetabolicAtlas/data-files", git_api)
+    for gem_name in gems_list:
+        repo_name = f"{owner}/{gem_name}"
+        gem_release_data = get_release_data(repo_name, git_api)
+        outfile = os.path.join(basedir, 'integrated-models', gem_name,
+                               'gemRepository.json')
+        if not is_dryrun:
+            writefile(json.dumps(gem_release_data, indent=2), outfile)
+            msg = f"Result file output to {outfile}"
+        else:
+            msg = f"DryRun: result file will be output to {outfile}"
+        print(msg)
+
+
+def main():
+    """main procedure"""
+    parser = argparse.ArgumentParser(
+        description='Generate data for GEMs repository',
+        formatter_class=argparse.RawDescriptionHelpFormatter
+        )
+    parser.add_argument('-d', '--dry', dest='is_dryrun', default=False,
+                        help='do not save output to file',
+                        action='store_true')
+
+    args = parser.parse_args()
+    is_dryrun = args.is_dryrun
+
+    try:
+        api_token = os.environ['GH_TOKEN']
+    except KeyError:
+        api_token = None
+
+    git_api = Github(api_token)
```

Comment:
It seems that we do not need a `GH_TOKEN`, I guess since this is a public repo? Maybe we can omit this section and just have `git_api = Github()` instead. We could also add a comment here instead explaining that it's not necessary. This is not a big deal and I can understand not making any changes here too. I think the code might be self-explanatory enough.

Reply:
There is a limit to access the API if token is not provided. I don't know the exact limit but it seems it's quite easy to hit the limit and then you will get a warning. Maybe we should even make the requirement for the TOKEN compulsory. What do you think @e0 and @mihai-sysbio ?

Reply:
We could also comment out the token part and re-enable it only when we hit the limit.

Reply:
This is actually how it works now. Token is not required by default but you will get a warning message when the limit is hit and the script will prompt you to enter a token.

Reply:
> This is actually how it works now.

doh! I should have read the code. I like how this is dealt with, so no need to change anything from my point of view.